tests/unit/createPackage/models/createPackageModel.spec.ts

```diff
@@ -52,316 +75,677 @@ describe('CreatePackageManager', () => {
-        dbId: layerFromCatalog.id,
-        zoomLevel: 7,
-        sanitizedBbox: expectedsanitizedBbox,
-        crs: userInput.crs as string,
-      };
-
-      findLayerMock.mockResolvedValue(layerFromCatalog);
-      createMock.mockResolvedValue(undefined);
-      updateJobMock.mockResolvedValue(undefined);
-      findCompletedJobMock.mockResolvedValue(undefined);
-      findInProgressJobMock.mockResolvedValue(JSON.parse(JSON.stringify(inProgressJob)));
-
-      const res = await createPackageManager.createPackage(userInput);
-      const expectedReturn: ICreateJobResponse = {
-        id: inProgressJob.id,
-        taskIds: [(inProgressJob.tasks as unknown as IJobResponse<IJobParameters, ITaskParameters>[])[0].id],
-        status: OperationStatus.IN_PROGRESS,
-      };
-
-      expect(res).toEqual(expectedReturn);
-      expect(findLayerMock).toHaveBeenCalledWith(jobDupParams.dbId);
-      expect(findLayerMock).toHaveBeenCalledTimes(1);
-      expect(createMock).toHaveBeenCalledTimes(0);
-      expect(findCompletedJobMock).toHaveBeenNthCalledWith(1, jobDupParams);
-      expect(findCompletedJobMock).toHaveBeenNthCalledWith(2, jobDupParams);
-      expect(findCompletedJobMock).toHaveBeenCalledTimes(2);
-      expect(findInProgressJobMock).toHaveBeenCalledWith(jobDupParams);
-      expect(findInProgressJobMock).toHaveBeenCalledTimes(1);
-      expect(findPendingJobMock).toHaveBeenCalledTimes(0);
+  });
+  describe('ROI', () => {
+    describe('#createExportMetadata', () => {
+      it('should create metadata.json file with the correct parameters', async () => {
+        const gpkgLocation = configMock.get<string>('gpkgsLocation');
+        const concatFsPathsSpy = jest.spyOn(utils, 'concatFsPaths');
+        const parseFeatureCollectionSpy = jest.spyOn(utils, 'parseFeatureCollection');
+        findLayerMock.mockResolvedValue(layerFromCatalogSample);
+
+        await createPackageManager.createExportJsonMetadata({ ...completedExportJob });
+        expect(fs.promises.writeFile).toHaveBeenCalledTimes(1);
+        expect(parseFeatureCollectionSpy).toHaveBeenCalledTimes(1);
+        expect(parseFeatureCollectionSpy).toHaveBeenCalledWith(completedExportJob.parameters.roi);
+        expect(concatFsPathsSpy).toHaveBeenCalledTimes(1);
+        expect(concatFsPathsSpy).toHaveBeenCalledWith(
+          gpkgLocation,
+          completedExportJob.parameters.relativeDirectoryPath,
+          completedExportJob.parameters.fileNamesTemplates.metadataURI
+        );
+        const expectedFileName = utils.concatFsPaths(
+          gpkgLocation,
+          completedExportJob.parameters.relativeDirectoryPath,
+          completedExportJob.parameters.fileNamesTemplates.metadataURI
+        );
+        expect(fs.promises.writeFile).toHaveBeenCalledWith(expectedFileName, JSON.stringify(metadataExportJson));
+      });
+
+      it('should failed on metadata.json file with the correct parameters', async () => {
```

Comment:
failed -> fail - because layer not found in catalog

Reply:
done